content/en/docs/03.0/3.3/openshift-images.md

```diff
@@ -0,0 +1,283 @@
+---
+title: "3.3 OpenShift images"
+linkTitle: "OpenShift images"
+weight: 33
+sectionnumber: 3.3
+description: >
+  This section is covering how to build a runnable image for OpenShift clusters.
+---
+
+## Application
+
+We use the same Go application from Lab 2 and extend it with a log file. Every request is logged with the actual time and the client IP address. The log file is placed under `/home/golang/hello-go.log`
+
+``` golang
+package main
+
+import (
+  "fmt"
+  "log"
+  "net/http"
+  "os"
+  "path/filepath"
+  "time"
+)
+
+var logFile = "/home/golang/hello-go.log"
+
+func main() {
+
+  _, err := createFile(logFile)
+  if err != nil {
+    panic(err)
+  }
+
+  http.HandleFunc("/", HelloServer)
+  http.ListenAndServe(":8080", nil)
+}
+
+func HelloServer(w http.ResponseWriter, r *http.Request) {
+  fmt.Fprintf(w, "Hello, %s!", r.URL.Path[1:])
+  appendToFile(r.RemoteAddr)
+}
+
+func createFile(p string) (*os.File, error) {
+  if err := os.MkdirAll(filepath.Dir(p), 0644); err != nil {
+    return nil, err
+  }
+  return os.Create(p)
+}
+
+func appendToFile(remoteAddr string) {
+  f, err := os.OpenFile(logFile, os.O_APPEND|os.O_WRONLY, 0644)
+  if err != nil {
+    log.Println(err)
+  }
+  defer f.Close()
+
+  line := fmt.Sprintf("%s: Request from: %s\n", time.Now().String(), remoteAddr)
+
+  if _, err := f.WriteString(line); err != nil {
+    log.Println(err)
+  }
+}
+
+```
+
+[source](https://raw.githubusercontent.com/puzzle/amm-techlab/master/content/en/docs/03.0/3.3/main.go)
+
+
+## Dockerfile
+
+The dockerfile is the same as in chapter 2.
+
+``` dockerfile
+FROM golang:1.14-alpine as builder
+COPY main.go /opt/app-root/src
+RUN env CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o go-hello-world-app .
+
+FROM alpine
+COPY --from=builder /opt/app-root/src/go-hello-world-app /home/golang/
+EXPOSE 8080
+ENTRYPOINT /home/golang/go-hello-world-app
+
+```
+
+[source](https://raw.githubusercontent.com/puzzle/amm-techlab/master/content/en/docs/03.0/3.3/Dockerfile)
+
+
+### Build
+
+Let's build the image with following command:
+
+
+```BASH
+buildah bud -f Dockerfile -t go-hello-world-os .
+```
+
+Check if the image is available locally
+
+```BASH
+buildah images
+```
+
+```
+REPOSITORY                                   TAG           IMAGE ID       CREATED         SIZE
+localhost/go-hello-world-os                  latest        d51386c82b80   2 minutes ago   7.42 MB
+localhost/go-hello-world                     latest        0493b6a22386   23 hours ago    218 MB
+```
+
+
+### Image test
+
+Test the container image locally using Podman. Use this command to run the container:
+
+```BASH
+podman run -p 8082:8080 -ti go-hello-world-os
+```
+
+Be aware of the new allocated port! (8082)
+
+```BASH
+curl localhost:8082/world
+```
+
+
+### Publish image to Dockerhub
```

Comment:
Good example! Ev. it's easier for the users to use Docker Builds on OpenShift. Then they only need to start a build again. First local test ist good to show that the Image works. What do you think?